### PR TITLE
Use UTF 8 instead of Latin-1

### PR DESCRIPTION
**Pull request by malthe**
This changes the hard-coded decoding of non-unicode strings to UTF-8.

### CODE REVIEW

**Comment by icemac** [comment]
@malthe Thank you for your PR. What is the rationale behind this PR? What problem are you trying to solve?

File: src/DocumentTemplate/cDocumentTemplate.c [hunk 1]
```diff
@@ -925,7 +925,7 @@ _join_unicode(PyObject *prejoin)
             PyObject *item = PyList_GetItem(list,i);
             if(PyString_Check(item))
             {
-                PyObject *unicode = PyUnicode_DecodeLatin1(PyString_AsString(item),PyString_Size(item),NULL);
+                PyObject *unicode = PyUnicode_DecodeUTF8(PyString_AsString(item),PyString_Size(item),NULL);
```
Comment (icemac) [inline src/DocumentTemplate/cDocumentTemplate.c]: There is a docstring for `join_unicode` in the same file which might have to be changed too.

**Comment by icemac** [comment]
@hannosch What do you think about this change?
If it only changes the way ids are decoded IMHO it should be not critical as they are currently ASCII only.
If it is used in other places it could be a problem as `latin-1` encoded strings cannot be decoded using `utf-8` thus it can break legacy Zope 2.13 installations on update.

**Comment by icemac** [comment]
@malthe I rebased your branch onto the current 2.13 branch to get Python 2.6 running on TravicCI.

**Comment by malthe** [comment]
A different path to take would be to make it configurable on a render-basis such that Zope can attach its `management_page_charset` option.

I just don't know if it's worth the bother.

File: src/DocumentTemplate/cDocumentTemplate.c [hunk 2]
```diff
@@ -960,7 +960,7 @@ static struct PyMethodDef Module_Level__methods[] = {
    "join a list of plain strings into a single plain string,\n"
    "a list of unicode strings into a single unicode strings,\n"
    "or a list containing a mix into a single unicode string with\n"
-   "the plain strings converted from latin-1"},
+   "the plain strings converted from UTF-8"},
```
Comment (tseaver) [inline src/DocumentTemplate/cDocumentTemplate.c]: If we do this, we need to update the [Python reference implementation](https://github.com/zopefoundation/DocumentTemplate/blob/cb27035473a610dc78c630669a29241935947d7a/src/DocumentTemplate/_DocumentTemplate.py#L140) as well.

Comment (malthe) [reply src/DocumentTemplate/cDocumentTemplate.c]: Updated just now in 1eedab1dbfa1c57f4e40b5397f207759ef8e0d01.

**Comment by malthe** [comment]
I suggest then that we opt for the second suggestion, which is to make the encoding configurable through ``render_blocks``:
```python
render_blocks(blocks, md, encoding=None)
```
If the encoding is `None` it will then pull from a module global:
```python
DEFAULT_ENCODING = "latin-1"
```
Then we can perhaps update the relevant parts in Zope 2 to connect this to `management_page_charset`.

**Comment by malthe** [comment]
I have implemented @tseaver's second suggestion here: https://github.com/zopefoundation/DocumentTemplate/pull/23.

**Pull request closed without merging**